api/server/router/distribution/backend.go

```diff
@@ -0,0 +1,18 @@
+package distribution
+
+import (
+	"github.com/docker/distribution"
+	"github.com/docker/distribution/reference"
+	"github.com/docker/docker/api/types"
+	"golang.org/x/net/context"
+)
+
+// Backend is all the methods that need to be implemented
+// to provide image specific functionality.
+type Backend interface {
+	distributionBackend
+}
+
+type distributionBackend interface {
```

Comment:
It's not necessary, but looking at other backends it seemed like convention to do this. I'm removing the private interface.